Mathlib/NumberTheory/Modular.lean

```diff
@@ -450,27 +447,20 @@ theorem abs_c_le_one (hz : z ∈ 𝒟ᵒ) (hg : g • z ∈ 𝒟ᵒ) : |g 1 0| 
     rcases eq_or_ne c 0 with (hc | hc)
     · rw [hc]; norm_num
     · refine (abs_lt_of_sq_lt_sq' ?_ (by norm_num)).2
-      specialize this hc
-      linarith
+      linear_combination this hc
   intro hc
-  replace hc : 0 < c ^ 4 := by
-    change 0 < c ^ (2 * 2); rw [pow_mul]; apply sq_pos_of_pos (sq_pos_of_ne_zero hc)
-  have h₁ :=
-    mul_lt_mul_of_pos_right
-      (mul_lt_mul'' (three_lt_four_mul_im_sq_of_mem_fdo hg) (three_lt_four_mul_im_sq_of_mem_fdo hz)
-        (by linarith) (by linarith))
-      hc
+  have h₁ : 3 * 3 * c ^ 4 < 4 * (g • z).im ^ 2 * (4 * z.im ^ 2) * c ^ 4 := by
```

Comment:
The changes in this file also look great to me!

Reply:
This golf works equally well without changing `linarith` to `linear_combination`; moved it to #19319.